core/blockchain/abci/abci.go

```diff
@@ -194,6 +195,161 @@ func AddCommonDeliverTxEvents(resp types.ResponseDeliverTx, tx Tx) types.Respons
 	return resp
 }
 
+const (
```

Comment:
Alright, I've just check with @edd and he doesn't need the details in here (e.g: StakeDeposited, StakeRemove, etc) in the ChainEvent, so all ChainEvent should have the type ChainEvent

Reply:
@jeremyletang right, however I think we still need to be able to filter by say `Deposit`, which can only be found in the `ChainEvent`. Otherwise, we already have filtering by command type, which in this case is incomplete.